### PR TITLE
[main > release/0.24.x]: Add support for office.com urls and fix driver to get corrrect id for tree.

### PR DESCRIPTION
**Pull request by jatgarg**
Cherry pick: 
1.) https://github.com/microsoft/FluidFramework/pull/2632
2.) https://github.com/microsoft/FluidFramework/pull/3633

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -569,16 +569,21 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
...
+                        if (odspSnapshot.trees.length > 0) {
+                            treeId = odspSnapshot.trees[0].id ?? "";
+                        }
                     }
                     if (odspSnapshot.blobs) {
                         this.initBlobsCache(odspSnapshot.blobs);
                     }
                 }
-                return this.treesCache.get(id);
+                // If the version id doesn't match with the id of the tree, then use the id of first tree which in that case
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:584]: I know it's port of prior changes, but some suggestions:
1. I'd not use treeId = "" above, but rather use undefined and not fetch from this.treesCache with empty string. Who knows, maybe in the future there will be some special entry with "" name.
2. I think this change says that there are two set of related IDs. Can we name them properly? I.e. id - what this ID is for? Just looking at the treatment, I'm not even sure that right API is called in this workflow. I.e. this is versionId, not treeID. Can this API be called with either version ID and tree ID?

**Review by vladsud — APPROVED** [review]

**Comment by vladsud** [comment]
BTW, you might want to consider adding my change that removes ability to start with ops, I think it's not in 0.24

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts]: Made treeId to be string or undefined. After talking to marcelo, we reached the conclusion that we should always fetch from the cache with the id of tree.

**Comment by jatgarg** [comment]
> BTW, you might want to consider adding my change that removes ability to start with ops, I think it's not in 0.24

Can you point to that change?

**Review by arinwt — APPROVED** [review]

**Pull request merged**